lib/releases.js

```diff
@@ -164,312 +213,435 @@ async function create_release(pg_pool, app_src, app_dest, build_uuid, descriptio
         if (release_update.status === 'pending') {
           console.log(`Error: The release never reported back as to what happened app_uuid: ${app_dest.app} release: ${release_id}`);
           await common.update_release_status(pg_pool, app_dest.app, release_id, 'unknown');
         }
       } catch (e) {
         // If the release doesn't exist its possible the app was removed before the release
         // timed out, this happens commonly in the tests, rare outside of that.
-        if(!(e instanceof common.NotFoundError)) {
-          console.error("Error updating release status while checking timeout: ", e, release)
+        if (!(e instanceof common.NotFoundError)) {
+          console.error('Error updating release status while checking timeout: ', e, release);
         }
       }
     }, 10 * 60 * 1000);
   } catch (e) {
     await common.update_release_status(pg_pool, app_dest.app, release_id, 'failed');
-    console.error("Error recording and requesting release:", e, release);
-    throw e
+    console.error('Error recording and requesting release:', e, release);
+    throw e;
   }
-  let release_event = {
-    'action':'release',
-    'app':{
-      'name':app_dest.name,
-      'id':app_dest.app,
+  const release_event = {
+    action: 'release',
+    app: {
+      name: app_dest.name,
+      id: app_dest.app,
     },
-    'space':{
-      'name':app_dest.space,
+    space: {
+      name: app_dest.space,
     },
-    'release':{
-      'id':release.id,
-      'result':'succeeded',
-      'created_at':release.created.toISOString(),
+    release: {
+      id: release.id,
+      result: 'succeeded',
+      created_at: release.created.toISOString(),
       version,
       description,
     },
-    'build':{
-      'id':build.id,
-      'result':'succeeded',
-      'repo':build.repo,
-      'commit':build.sha,
-      'branch':build.branch,
-    }
-  }
-  logs.event(pg_pool, app_dest.name, app_dest.space, "Release v" + version + " created (" + description + ")");
-  common.lifecycle.emit('release', release_event)
-  common.notify_hooks(pg_pool, app_dest.app, 'release', JSON.stringify(release_event), user ? user : "System")
-  return release_obj_to_response(release)
+    build: {
+      id: build.id,
+      result: 'succeeded',
+      repo: build.repo,
+      commit: build.sha,
+      branch: build.branch,
+    },
+  };
+  logs.event(pg_pool, app_dest.name, app_dest.space, `Release v${version} created (${description})`);
+  common.lifecycle.emit('release', release_event);
+  common.notify_hooks(pg_pool, app_dest.app, 'release', JSON.stringify(release_event), user || 'System');
+  return release_obj_to_response(release);
 }
 
-async function create(pg_pool, app_uuid, app_name, space_name, space_tags, org, description, slug, release_id, trigger_notes, user) {
-  if(!slug && !release_id) {
-    throw new common.UnprocessibleEntityError(`The specified "slug" (or) "release" field was not provided.`)
+async function create(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Async function 'create' has a complexity of 11. Maximum allowed is 8.](https://app.codacy.com/manual/Akkeris/controller-api/pullRequest?prid=5244943)